react/components/EditorContainer/Sidebar/ComponentList/index.tsx

```diff
@@ -52,10 +46,12 @@ class ComponentList extends Component<Props, State> {
   public static getDerivedStateFromProps(props: Props, state: State) {
     if (!equals(props.components, state.initialComponents)) {
       return {
+        ...state,
         components: normalize(props.components),
         initialComponents: props.components,
       }
     }
+    return state
```

Comment:
I think that the return value should be `null` when state updates aren't needed. I'm not sure, but I believe that returning something other than `null` may trigger unnecessary re-renders.